packages/trader/src/AppV2/Components/ContractDetailsFooter/contract-details-footer.tsx

```diff
@@ -0,0 +1,84 @@
+import { Button } from '@deriv-com/quill-ui';
+import { TContractInfo, getCardLabels, isMultiplierContract, isValidToCancel, isValidToSell } from '@deriv/shared';
+import { useStore } from '@deriv/stores';
+import { getRemainingTime } from 'AppV2/Utils/helper';
+import { observer } from 'mobx-react';
+import React from 'react';
+
+type ContractInfoProps = {
+    contract_info: TContractInfo;
+};
+
+const ContractDetailsFooter = observer(({ contract_info }: ContractInfoProps) => {
+    const {
+        bid_price,
+        currency,
+        contract_id,
+        cancellation: { date_expiry: cancellation_date_expiry } = {},
+        profit,
+        contract_type,
+    } = contract_info;
+
+    const { contract_replay, common } = useStore();
+    const { server_time } = common;
+    const { onClickCancel, onClickSell, is_sell_requested } = contract_replay;
+    const is_valid_to_sell = isValidToSell(contract_info);
+    const is_valid_to_cancel = isValidToCancel(contract_info);
+    const is_multiplier = isMultiplierContract(contract_type);
+
+    const cardLabels = getCardLabels();
+    const bidDetails = !is_valid_to_cancel ? `@${bid_price} ${currency}` : '';
+    const label = `${cardLabels.CLOSE} ${bidDetails}`;
+
+    return (
+        <div className='contract-details-footer--container'>
+            {is_multiplier ? (
+                <>
+                    <Button
+                        variant='secondary'
+                        label={label}
+                        color='black'
+                        size='lg'
+                        isLoading={is_sell_requested}
+                        disabled={is_sell_requested || (Number(profit) < 0 && is_valid_to_cancel)}
+                        onClick={() => onClickSell(contract_id)}
+                        fullWidth
+                    />
+                    {is_valid_to_cancel && (
+                        <Button
+                            onClick={() => onClickCancel(contract_id)}
+                            variant='secondary'
+                            label={`${cardLabels.CANCEL} ${getRemainingTime({
+                                end_time: cancellation_date_expiry,
+                                format: 'mm:ss',
+                                getCardLabels,
+                                start_time: server_time,
+                            })}`}
```

Comment:
please use `<RemainingTime />` component from `@deriv/components` instead of creating this new `getRemainingTime()` helper.
The logic inside `<RemainingTime />` component is error-proof and we also used it on our new Positions page as well.

Reply:
Thanks added